lxqtsettings.cpp

```diff
@@ -214,7 +246,10 @@ void Settings::_fileChanged(QString path)
     // Luckily, I found a workaround: If the file path no longer exists
     // in the watcher's files(), this file is deleted.
     if(!d_ptr->mWatcher.files().contains(path))
-        d_ptr->mWatcher.addPath(path);
+        // in some situations adding fails because of non-existing file (e.g. editting file in external program)
+        if (!d_ptr->mWatcher.addPath(path) && 0 == d_ptr->mAddWatchTimer)
+            d_ptr->mAddWatchTimer = startTimer(100);
```

Comment:
It seems to me we may get into a infinte loop here if the file gets deleted. Adding the file won't be work, so the timer will be started, everything again. Tell me I'm wrong.